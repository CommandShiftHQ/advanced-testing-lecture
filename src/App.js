import React from "react";
import moment from "moment";
import "./App.css";

function App(props) {
  const weatherReportForToday = props.weatherData;

  return (
    <div className="App">
      <div className="date" data-testid="date-id">
        {moment(weatherReportForToday.date).format("ddd Do MMM")}
      </div>
      <div className="temperature" data-testid="temperature-id">
        {weatherReportForToday.temperature.max}&deg;c
      </div>
      <div className="description" data-testid="description-id">
        {weatherReportForToday.description}
      </div>
    </div>
  );
}

export default App;
