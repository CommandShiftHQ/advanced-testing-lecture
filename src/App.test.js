import React from 'react';
import { render, getByTestId } from '@testing-library/react';
import App from './App';

const TEST_WEATHER_DATA = {
  date: 1525046400000,
  temperature: {
    max: 11,
    min: 4,
  },
  wind: {
    speed: 13,
    direction: "s",
  },
  humidity: 30,
  description: "Clear",
  icon: "800",
};

// best effort
test('component renders', () => {
  const { asFragment } = render(<App weatherData={TEST_WEATHER_DATA} />);
  const component  = asFragment();
  // when using snapshots always open the resulting snapshot file
  // and visually check if the render is what you expect it to be!
  expect(component).toMatchSnapshot(); 
});


// you might just check test is in the document
// naive effort
test('renders weather status', () => {
  const { getByText } = render(<App weatherData={TEST_WEATHER_DATA} />);
  const weatherStatus = getByText(/Clear/i);
  expect(weatherStatus).toBeInTheDocument();
});

// manually check each section is rendered correctly
// highly specific tests 
// a lot of effort!
test('renders weather status', () => {
  const { getByTestId } = render(<App weatherData={TEST_WEATHER_DATA} />);
  const date = getByTestId("date-id");
  expect(date).toHaveTextContent("Mon 30th Apr");

  const temperature = getByTestId("temperature-id");
  expect(temperature).toHaveTextContent("11Â°c");

  const description = getByTestId("description-id");
  expect(description).toHaveTextContent("Clear");
});