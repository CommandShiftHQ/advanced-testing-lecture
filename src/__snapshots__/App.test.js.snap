// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders 1`] = `
<DocumentFragment>
  <div
    class="App"
  >
    <div
      class="date"
      data-testid="date-id"
    >
      Mon 30th Apr
    </div>
    <div
      class="temperature"
      data-testid="temperature-id"
    >
      11Â°c
    </div>
    <div
      class="description"
      data-testid="description-id"
    >
      Clear
    </div>
  </div>
</DocumentFragment>
`;
